src/views/AddItem.jsx

```diff
@@ -16,14 +16,58 @@ export function AddItem({ listToken }) {
 	const handleSubmit = async (e) => {
 		e.preventDefault();
 
+		//Functions solution for each error Message
+		// const isValid = (itemName) => {
+		// 	checkEmpty(itemName);
+		// 	checkDuplicates(itemName);
+		// }
+
+		// const checkEmpty =(itemName) => {
+		// 	itemName.replace(/\s/g, '').length === 0;
+		// 	setMessage("Error: Please enter your item's name.");
+		// }
+
+		// const itemNames = itemList.map((item) =>
+		// item.name.toLowerCase().replace(/[^a-zA-Z0-9 ]/g, ''),); // Item names without special characters for comparison
+
+		// const checkDuplicates =(itemNames) => {
+		// 	itemNames.includes(itemName.toLowerCase().replace(/[^a-zA-Z0-9 ]/g, ''));
+		// 	setMessage(`${itemName} is already on your list. Please add a different item.`,);
+		// }
+
+		// try {
+		// 	isValid();
+		// 	addItem(listToken, {
+		// 		itemName,
+		// 		daysUntilNextPurchase,
+		// 	});
+		// 	setMessage(`${itemName} was successfully saved to your shopping list!`);
+		//  } catch (error) {
+		//  	setMessage('Error: This item not added to the database.');
+		//  }
```

Comment:
Please delete these commented out lines if you don't need them any more :slightly_smiling_face:

Reply:
This commented part is the one I reworked according to Peter' suggestions but it is somehow not working. Could we check it today?